src/librustc_resolve/lib.rs

```diff
@@ -1965,7 +2072,7 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
             }
             if ns == TypeNS && is_known_tool(ident.name) {
                 let binding = (Def::ToolMod, ty::Visibility::Public,
-                               ident.span, Mark::root()).to_name_binding(self.arenas);
+                               DUMMY_SP, Mark::root()).to_name_binding(self.arenas);
```

Comment:
In this case we need a *definition* span, which is dummy if the thing is defined implicitly by the language, e.g. it's a built-in type or implicitly injected extern crate std.
`ident.span` in this context would mean that built-in types are defined-on-use multiple times.

`binding_description` indeed looks at spans and describes things with dummy spans as built-in or from prelude.

(Also, I want to pre-intern `NameBinding`s for the closed set of built-in things eventually, instead of allocating new clones on every use, so they need to have identical data including span.)